src/battle_script_commands.c

```diff
@@ -10960,6 +10994,7 @@ static void Cmd_setsemiinvulnerablebit(void)
     {
     case MOVE_FLY:
     case MOVE_BOUNCE:
+    case MOVE_SKY_DROP:
         gStatuses3[gBattlerAttacker] |= STATUS3_ON_AIR;
```

Comment:
set the on air bit on the target as well, because the target is also in air

Reply:
This is handled within Cmd_various function in this file, with the VARIOUS_SET_SKY_DROP and VARIOUS_CLEAR_SKY_DROP cases, as there are a few other things that need to be handled when the target is affected by Sky Drop. I also feel that it is unwise to add code that pertains to only one particular event to a function that is used for multiple other events, at least when it can be reasonably avoided, which it can in this situation.

Reply:
ah you're right, i didn't realize that, my bad